### DISCUSSION TITLE
How to use `@request.*` filter fields?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#630] [Q&A] [answered]
Hey, how to use `@request.*` filter to access request data?

**Comment by ganigeorgiev** [comment, marked as answer]
In short, you have access to the following request fields:

- `@request.method` - the HTTP request method (eg. `@request.method = 'GET'`)
- `@request.query.*` - the request query parameters (eg. `@request.query.page = 1`)
- `@request.data.*` - the submitted body parameters (eg. `@request.data.title != ""`)
- `@request.user.*` - the current authenticated user (eg. `@request.user.id != '' && @request.user.profile.role = 'member'`)

You could also find some examples under the Collection API rules panel -> "Show available fields":

![rules_screenshot](https://pocketbase.io/images/screenshots/collection-rules.png)
